storage/table.go

```diff
@@ -40,64 +33,55 @@ type TableAccessPolicy struct {
 	CanDelete  bool
 }
 
-func pathForTable(table AzureTable) string { return fmt.Sprintf("%s", table) }
-
-func (c *TableServiceClient) getStandardHeaders() map[string]string {
-	return map[string]string{
-		"x-ms-version":   "2015-02-21",
-		"x-ms-date":      currentTimeRfc1123Formatted(),
-		"Accept":         "application/json;odata=nometadata",
-		"Accept-Charset": "UTF-8",
-		"Content-Type":   "application/json",
-		userAgentHeader:  c.client.userAgent,
-	}
+// Table represents an Azure table.
+type Table struct {
+	tsc            *TableServiceClient
+	Name           string `json:"TableName"`
+	OdataEditLink  string `json:"odata.editLink"`
+	OdataID        string `json:"odata.id"`
+	OdataMetadata  string `json:"odata.metadata"`
+	OdataType      string `json:"odata.type"`
+	AccessPolicies []TableAccessPolicy
 }
 
-// QueryTables returns the tables created in the
-// *TableServiceClient storage account.
-func (c *TableServiceClient) QueryTables() ([]AzureTable, error) {
-	uri := c.client.getEndpoint(tableServiceName, tablesURIPath, url.Values{})
-
-	headers := c.getStandardHeaders()
-	headers["Content-Length"] = "0"
-
-	resp, err := c.client.execInternalJSON(http.MethodGet, uri, headers, nil, c.auth)
-	if err != nil {
-		return nil, err
-	}
-	defer resp.body.Close()
-
-	if err := checkRespCode(resp.statusCode, []int{http.StatusOK}); err != nil {
-		return nil, err
-	}
-
-	buf := new(bytes.Buffer)
-	if _, err := buf.ReadFrom(resp.body); err != nil {
-		return nil, err
-	}
-
-	var respArray queryTablesResponse
-	if err := json.Unmarshal(buf.Bytes(), &respArray); err != nil {
-		return nil, err
-	}
+// EntityQueryResult contains the response from
+// QueryEntities and QueryEntitiesNextResults functions.
+type EntityQueryResult struct {
+	OdataMetadata string   `json:"odata.metadata"`
+	Entities      []Entity `json:"value"`
+	NextLink      *string
+}
 
-	s := make([]AzureTable, len(respArray.TableName))
-	for i, elem := range respArray.TableName {
-		s[i] = AzureTable(elem.TableName)
-	}
+type continuationToken struct {
+	NextPartitionKey string
+	NextRowKey       string
+}
 
-	return s, nil
+func (t *Table) buildPath() string {
+	return fmt.Sprintf("/%s", t.Name)
 }
 
-// CreateTable creates the table given the specific
-// name. This function fails if the name is not compliant
+// Create creates the referenced table.
+// This function fails if the name is not compliant
 // with the specification or the tables already exists.
-func (c *TableServiceClient) CreateTable(table AzureTable) error {
-	uri := c.client.getEndpoint(tableServiceName, tablesURIPath, url.Values{})
+// See https://docs.microsoft.com/rest/api/storageservices/fileservices/create-table
+func (t *Table) Create(getResponse bool) error {
```

Comment:
Actually I think a `type AcceptMetadata string` would be better, so it works better with how the API works.